diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
index 264027e..73f022e 100644
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -2290,13 +2290,7 @@ set_define("_LIBCPP_HIDE_FROM_ABI", libcxx_override_visibility.hide_from_abi)
 @depends(target, build_environment)
 def visibility_flags(target, env):
     if target.os != "WINNT":
-        if target.kernel == "Darwin":
-            return ("-fvisibility=hidden", "-fvisibility-inlines-hidden")
-        return (
-            "-I%s/system_wrappers" % os.path.join(env.dist),
-            "-include",
-            "%s/config/gcc_hidden.h" % env.topsrcdir,
-        )
+        return ("-fvisibility=hidden", "-fvisibility-inlines-hidden")
 
 
 @depends(target, visibility_flags)
